rust/parser/mod.rs

```diff
@@ -49,279 +46,357 @@ use crate::{
         TypeQLUndefine, TypeQLUpdate,
     },
 };
-use antlr_rust::{
-    token::Token,
-    tree::{ParseTree, TerminalNode as ANTLRTerminalNode},
-};
 use chrono::{NaiveDate, NaiveDateTime};
-use std::rc::Rc;
+use pest::Parser;
+use pest_derive::Parser;
 
-// keep star import to not expose generated code
-use typeql_grammar::typeqlrustparser::*;
+#[derive(Parser)]
+#[grammar = "parser/typeql.pest"]
+struct TypeQLParser;
 
-type TerminalNode<'a> = ANTLRTerminalNode<'a, TypeQLRustParserContextType>;
+type SyntaxTree<'a> = pest::iterators::Pair<'a, Rule>;
+type Children<'a> = pest::iterators::Pairs<'a, Rule>;
 
-fn get_string(string: Rc<TerminalNode>) -> String {
-    unquote(&string.get_text())
+trait TreeChildren {
+    type Items;
+    fn into_children(self) -> Self::Items;
 }
 
-fn get_regex(string: Rc<TerminalNode>) -> String {
-    unescape_regex(&unquote(&string.get_text()))
+impl<'a> TreeChildren for SyntaxTree<'a> {
+    type Items = Children<'a>;
+    fn into_children(self) -> Self::Items {
+        self.into_inner()
+    }
 }
```

Comment:
Aliased from pest's `Pair`, `Pair::into_inner()`, and `Pairs` to more meaningful `SyntaxTree`, `SyntaxTree::into_children()` and `Children`